Adding Quota APIs to Capacity RP.

rahuls-microsoft opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

rahuls-microsoft committed:
Adding swagger Spec for Quota RP API.

rahuls-microsoft committed:
Adding Quota RP swagger API.

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-python

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        ],
+        "summary": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "description": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "operationId": "Quota_Check",
```
Operation ID should follow `NOUN_VERB` pattern where the noun is plural, so `Quotas_Check`.

rahuls-microsoft replied:
will update it.

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        ],
+        "summary": "Submits a Quota Request for a resource provider at the specified location.",
+        "description": "Submits Quota change request for a resource provider for the specified location. To use, first make a request to the /usages API to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which require an update to their quotas, update their limit fields in the response from the / ...
+        "operationId": "Quota_Request_Submit",
```
Operation IDs are used to generate class+method; i.e. left of first `_` is the class name and to the right is method name.  Also, from the description (and since this is a PUT) it sounds like this API can create a new or update an existing resource?  If so then the operation ID should be `QuotaRequests_CreateOrUpdate`.

rahuls-microsoft replied:
I can change it to - QuotaRequests_Create. There is no Update operation to QuotaRequest.

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        ],
+        "summary": "Deletes the QuotaRequest by name for the resources for the resource  provider at a specific location.",
+        "description": "For the specified location, deletes the quota request by request name for resources under the subscription. Delete will not cancel the Quota Request in progress. It will only delete the completed requests, whican be rejected or succeeded quota requests.",
+        "operationId": "QuotaRequests_Name_Delete",
```
`QuotaRequests_Delete`, or if there's more than one way to delete it then `QuotaRequests_DeleteByName`

rahuls-microsoft replied:
I will update to QuotaRequests_Delete.

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+          "200": {
+            "description": "Ok. Quota Request deleted."
+          },
+          "404": {
```
Excluding `default`, all status codes listed in the `responses` block are success codes.  If 404 and 409 are not supposed to be treated as API success please remove them.

rahuls-microsoft replied:
404 and 409 are API success code.
Denoting the user that the request was not found or cannot be removed.

ryansbenson replied:
You should only have 200 and 204 in here as success codes, 4xx are user errors not success and should be covered by 'default'.

rahuls-microsoft replied:
I will remove 4xx response.

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+              "$ref": "#/definitions/QuotaRequestDetailsByNameList"
+            }
+          },
+          "404": {
```
Success code?

rahuls-microsoft replied:
Yes

ryansbenson replied:
same comment, I don't see any way 404 is success. You may be thinking of the 204 pattern we use as 'success', take a look at this doc:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+          "QuotaRequests"
+        ],
+        "summary": "Deletes the QuotaRequest by name for the resources for the resource  provider at a specific location.",
+        "description": "For the specified location, deletes the quota request by request name for resources under the subscription. Delete will not cancel the Quota Request in progress. It will only delete the completed requests, whican be rejected or succeeded quota requests.",
```
```suggestion
        "description": "For the specified location, deletes the quota request by request name for resources under the subscription. Delete will not cancel the Quota Request in progress. It will only delete the completed requests, which can be rejected or succeeded quota requests.",
```

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        ],
+        "summary": "For the specified location and resource Provider, gets thecurrent quota requests under the subscription over the time\n period of one year ago from now to one year back.",
+        "description": "For the specified location and Resource provider gets the current quota requests under the subscription over the time period of one year ago from now to one year back. oData filter can be used to select quota requests.",
+        "operationId": "QuotaRequest_status",
```
`QuotaRequests_List`.  Should also have [x-ms-pageable](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable) attribute.

rahuls-microsoft replied:
will add it.

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotarequests": {
```
```suggestion
    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests": {
```

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        "tags": [
+          "QuotaRequests"
+        ],
+        "summary": "For the specified location and resource Provider, gets thecurrent quota requests under the subscription over the time\n period of one year ago from now to one year back.",
```
```suggestion
        "summary": "For the specified location and resource Provider, gets the current quota requests under the subscription over the time\n period of one year ago from now to one year back.",
```

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+          "200": {
+            "description": "Ok. Sending Quota request status and details",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestDetailsByNameList"
```
`QuotaRequestDetailsList`, unless there are really different lists based on how they were retrieved.

rahuls-microsoft replied:
QuotaRequestDetailsByName - Include the QuotaRequest Name also.
QuotaRequestDetail is response for - Get /subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName}
The request name is already specified in the request, so it's not included in the response.
Also changed QuotaRequestDetails to QuotaRequestDetail .
Removed QuotaRequestDetails

rahuls-microsoft replied:
Please review the changes and comments. I have addressed the questions and updted the API.

rahuls-microsoft replied:
Removed the error word from the response.

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```
Why is this set to null?  The response type defines a `nextLink` field.

rahuls-microsoft replied:
/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName} is for single request.
I will update the response to quotaRequestDetail and will remove QuotaRequestDetails.

rahuls-microsoft replied:
Updated the response.

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+        ],
+        "summary": "For the specified location, gets the Quota request status by request name provided during the quota request creation for the specific resources under the subscription.",
+        "description": "Gets the QuotaRequest details and status by the quota Request Name for the resources for the resource  provider at a specific location.",
+        "operationId": "QuotaRequests_Status_Check",
```
The summary is confusing, sounds like only one item is returned but the model indicates otherwise, can you please clarify?  If it's singular then it should be `QuotaRequests_Get`.

rahuls-microsoft replied:
That's right, only one item is returned.
Updated the response type to quotaRequestDetail. Please check the previous comment.
Updated OP-ID to QuotaRequests_Get

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

jhendrixMSFT commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,609 @@
+          }
+        ],
+        "responses": {
+          "202": {
```
Since this is an async operation you should add [x-ms-long-running-operation](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation).

rahuls-microsoft replied:
Sure will update it.
The operation returns immediately with the requestId.
The user can check the status with the GET /subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName}

jhendrixMSFT reviewed (changes requested):
Please add a README.md file, see https://github.com/Azure/azure-rest-api-specs/blob/master/specification/redis/resource-manager/readme.md for an example.

rahuls-microsoft committed:
Updated based on the comments.

rahuls-microsoft committed:
Updating based on the comments from jhendrixMSFT

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,595 @@
+          "QuotaRequests"
+        ],
+        "summary": "Deletes the QuotaRequest by name for the resources for the resource  provider at a specific location.",
+		"description": "For the specified location, deletes the quota request by request name for resources under the subscription. Delete will not cancel the Quota Request in progress. It will only delete the completed requests, whican be rejected or succeeded quota requests.",
```
```suggestion
        "description": "For the specified location, deletes the quota request by request name for resources under the subscription. Delete will not cancel the Quota Request in progress. It will only delete the completed requests, which can be rejected or succeeded quota requests.",
```

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,595 @@
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextpage"
```
If the camelCase doesn't make sense, you can add the word to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt to fix the spellcheck failure
```suggestion
            "nextLinkName": "nextPage"
```

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

rahuls-microsoft committed:
Updating quota request status parameter.

rahuls-microsoft committed:
Making all response for all quotaRequests status response same, for one or collection.

rahuls-microsoft committed:
Adding Filter for quotaRequest by requestSubmitTime.

rahuls-microsoft committed:
Fixing casing of the operationId - QuotaRequest_Status

rahuls-microsoft committed:
Removing unused model and adding initial version of readme.md

rahuls-microsoft committed:
Adding Readme.md File.

rahuls-microsoft committed:
Updated path for quota.json.

rahuls-microsoft committed:
Chaning filename Quota.json to quota.json in readme.ms. Looks like it's
case-sensitive.

rahuls-microsoft committed:
Fixing json.

rahuls-microsoft committed:
Updating based on build errors.

rahuls-microsoft committed:
Updating suppression directive for opeartions.

rahuls-microsoft committed:
Fixing Suppression Directive.

rahuls-microsoft committed:
Fixing List operationId name.

nschonni commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,591 @@
+          "QuotaInformation"
+        ],
+        "summary": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "description": "This Api gets the current quota limit and usages for the resources provider for the specified location. This response can be used to submit quoutaRequests.",
```
quouta -> quota
```suggestion
        "description": "This API gets the current quota limit and usages for the resources provider for the specified location. This response can be used to submit quotaRequests.",
```

ryansbenson commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,591 @@
+        }
+      }
+    },
+    "CreateGenericQuotaRequestParameters": {
```
The way you modeled the paths implies this is a resource and should be modeled as such with id, name, type, properties...

rahuls-microsoft replied:
I will remove 4xx response code.

rahuls-microsoft replied:
The get request is - /subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{location}/usages
To get the current resource usage and limit (quota).
The Put is quotaRequests, which as same as you described - change the limit field to requested quota request. 
The name doesn't match, I agree.
AS quota is not a resource, it's an operation on resources provide by different RPs- such as - compute, batchAI, etc.
That's why, we followed the /usages pattern to get the resource information and used /quotaRequests for Put quotRequests an the Get /quotaRequests is just checking the status of the quota requests.
Please let me know, if it sounds fine?

rahuls-microsoft replied:
Updated the API to match Get /serviceLimits and Put /ServiceLimits

ryansbenson commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,591 @@
+          "200": {
+            "description": "Ok. Returning the status for all relevant quota requests .",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestDetailsByName"
```
The schema for put input should be the same as the get response. In this case: CreateGenericQuotaRequestParameters.

Can these types be combined?

rahuls-microsoft replied:
The reason explained above.
The put /quotaRequests is constructed from get /usages . They are the pair for get and put.

rahuls-microsoft replied:
Updated the API to match Get /serviceLimits and Put. Used the same data model.

rahuls-microsoft replied:
The wrapper is changed. The get response has  "nextLink": "string". 
The Put shouldn't have  "nextLink": "string".
Any suggestions?

rahuls-microsoft replied:
Put doesn't have nextLink, as the whole set of subrequests is treated as single request.

ryansbenson commented on specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json:
```diff
@@ -0,0 +1,591 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{location}/usages": {
```
Can you add examples for all these APIs?

rahuls-microsoft replied:
I will add examples.

rahuls-microsoft committed:
Update specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

rahuls-microsoft committed:
1. Updated the _list_ as suggested in the build.
2. Removed the 4XX responses.

rahuls-microsoft committed:
Adding example file.

rahuls-microsoft committed:
1.Adding providerId in the parameter List.

rahuls-microsoft committed:
1. Changed the namesace to Microsoft.Capacity. Work with Capacity RP Team (Reservation) to share the namespace.
2. Updated the API to Get and Put pair. Use the same model for Get and Put as suggested.

Todo:
1. Once Apis are finalized add samples for each api.

rahuls-microsoft committed:
1. Fixing spelling.
2. Updateing the Put datamodel to match Get data model.

rahuls-microsoft committed:
rename QuotaRequests_list_Status - QuotaRequests_listStatus.
There was a warning for QuotaRequests_list_Status.

rahuls-microsoft committed:
1. Updated API.
2. Added example for Put /serviceLimits.
3. Updated Get example - /serviceLimits.

rahuls-microsoft committed:
1. Adding examples for /quotaRequests/{id}
2. Adding examples for /quotaRequests

rahuls-microsoft committed:
1. Formatting changes.
2. Fixing example.

rahuls-microsoft committed:
1. Removing nextLink from get /serviceLimits, as the RPs doesn't support nextLink for /usages.

rahuls-microsoft committed:
1. Moving the examples, next to the operationId.
2. Making the arrary "value" instead of quotaRequests.

rahuls-microsoft committed:
1. Changing quotaRequests to value.

rahuls-microsoft committed:
Fixing example filed name to match the model.

rahuls-microsoft committed:
1. Fixing example for Put request.
2. Supressing warning PageableOperation or Get /serviceLimits, as all the RPs /usages api doesn't support pagable operation.

rahuls-microsoft committed:
Fixing readme.md

rahuls-microsoft committed:
1. Removing pageableOperation Suppression.
2. Removing readonly roperties from put parameters.

rahuls-microsoft committed:
1. Changing quotaRequests to value to address - warning PutRequestResponseScheme.
2. Suppressing RequiredPropertiesMissingInResourceModel. The quotaRequest doesn't have the parameters specified.

rahuls-microsoft committed:
1. Making Put request and response same to address PutRequestResponseScheme

rahuls-microsoft committed:
1. Changing requestId to id to address BodyTopLevelProperties

rahuls-microsoft committed:
Adding example for BatchAI /serviceLimits

rahuls-microsoft committed:
1. Adding the swagger definition files for Quota.
2. Updating Readme.md to include Quota.json
3. Updating the title to "Microsoft.Capacity", it's required to be same in all the files.

rahuls-microsoft committed:
Updating readme.md to resolve conflicy with master.

rahuls-microsoft committed:
Removing the conflict.

rahuls-microsoft committed:
Merge branch 'master' into rahuls-quota-spec

# Conflicts:
#	specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

rahuls-microsoft committed:
Removing space, which was causing merge conflict.

rahuls-microsoft committed:
Removing files from Quota path.
The Quota APIs will be hosted with Microsoft.Capacity RP.

rahuls-microsoft committed:
1. Addressing SDK warning for - XmsResourceInPutResponse.

rahuls-microsoft committed:
Updating QuotaRequestResponse Model.

rahuls-microsoft committed:
Updating examples, based on the model change.

rahuls-microsoft committed:
Model change.

rahuls-microsoft committed:
Model Update.

rahuls-microsoft committed:
Adding description.

rahuls-microsoft committed:
Updated example

rahuls-microsoft committed:
1. Validating new APIs.
2. Will add examples in next iteration.

rahuls-microsoft committed:
1. Updating AI and adding examples.

rahuls-microsoft committed:
1. Changing Operation_id

rahuls-microsoft committed:
1. Added examples.
2. Added properties object to hold additional properties from different resource providers.

rahuls-microsoft committed:
Adding examples for QuotaRequests.

rahuls-microsoft committed:
Adding missing example file.
adding x-ms-azure-resource : true

rahuls-microsoft committed:
Setting - x-ms-azure-resource : true

rahuls-microsoft committed:
Adding example for MsSql get usages.
Fixing few warnings.

rahuls-microsoft committed:
Setting "x-ms-pageable": {
          "nextLinkName": "nextLink"
        }

for the APIs, which are not pageable, returns only one object.

rahuls-microsoft committed:
Removing extra ,

yungezz commented:
hi @ryansbenson , could you pls review this PR again? thanks.

rahuls-microsoft committed:
Updating name to resourceLimits

rahuls-microsoft committed:
Adding example of quotaRequest in progress and quotaRequest Failed.

rahuls-microsoft committed:
Changing name back to serviceLimits.

rahuls-microsoft committed:
Removing "x-ms-client-flatten": true, from the propties,
which may hold specific properties for the RP.

yungezz commented:
hi @rahuls-microsoft the above error is fixed. and there's error in your operationId , pls fix the lintdiff error https://dev.azure.com/azure-sdk/public/_build/results?buildId=98261&view=logs&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d.

rahuls-microsoft committed:
1. Fixing errors - Reming Nulls for next page link.
2. The get request for 1 item doesn't need to be pageable.

rahuls-microsoft commented:
Looks like there are these 2 potential erros:

Potential new ARM errors


========================

R2020 - RequiredPropertiesMissingInResourceModel

Model definition 'QuotaRequestDetails' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.

  at specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:550



R3006 - BodyTopLevelProperties

Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'QuotaRequestDetails' has extra properties ['value'].

  at specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:553

rahuls-microsoft commented:
I have Id property and other properties wrapped in an object : quotaRequestProperties. Those properties are specific for quotaRequest. So, we need it.
How do I address it?
Thanks.
Rahul.

rahuls-microsoft committed:
1.Updating the name for ResourceType

rahuls-microsoft committed:
1. Adding Auto Quota Request API.

rahuls-microsoft committed:
1. Updated AQI Api.

rahuls-microsoft committed:
Fixing recipients spelling.
Removing x-ms-azure-resource

rahuls-microsoft committed:
1. Adding examples for Auto Quota Increase.

rahuls-microsoft committed:
1. Adding example for turning off th Auto Quota Invrease.
2. Fixing examples.

rahuls-microsoft committed:
Updating the model.

rahuls-microsoft committed:
Fixing reference.

rahuls-microsoft committed:
Fixing spelling of Sev

rahuls-microsoft committed:
Adding "x-ms-azure-resource": true,

rahuls-microsoft committed:
Adding name, type parameters.

rahuls-microsoft committed:
Fixing top level elements.

rahuls-microsoft commented:
Addressed all the errors.
Please take a look.

Is this error due to some system issue:
autoRest core      '/root/.autorest/@microsoft.azure_autorest-core@2.0.4390/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4390)
INFO:swaggertosdk.autorest_tools:   Installing AutoRest extension '@microsoft.azure/autorest.python' (~4.0.71)
INFO:swaggertosdk.autorest_tools:   Installing AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.44)
INFO:swaggertosdk.autorest_tools:FATAL: System.InvalidOperationException: Sequence contains no elements
INFO:swaggertosdk.autorest_tools:   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
INFO:swaggertosdk.autorest_tools:   at AutoRest.Python.Model.MethodPy.BuildInputMappings() in /home/vsts/work/1/s/src/vanilla/Model/MethodPy.cs:line 648
INFO:swaggertosdk.autorest_tools:   at AutoRest.Python.azure.Templates.AzureLongRunningMethodTemplate.<ExecuteAsync>d__1.MoveNext() in /home/vsts/work/1/s/src/obj/Razor/azure/Templates/AzureLongRunningMethodTemplate.cshtml:line 17
INFO:swaggertosdk.autorest_tools:--- End of stack trace from previous location where exception was thrown ---

Is there any action needed from my side?
Thanks.
Rahul.

rahuls-microsoft committed:
Updating descriptions and summary text.

rahuls-microsoft committed:
Updating Id with full resource path.

rahuls-microsoft committed:
Speeling fix for specified

rahuls-microsoft committed:
Removing '' from the json.

rahuls-microsoft committed:
,

raych1 commented:
@zikalino, can you have a look at the python sdk errors?
```code
ERROR:swaggertosdk.autorest_tools:Command '['/usr/local/bin/autorest', '/git-restapi/specification/reservations/resource-manager/readme.md', '--keep-version-file', '--multiapi', '--no-async', '--python', '--python-mode=update', '--python-sdks-folder=/tmp/tmpzdrg8__9/sdk/sdk', '--use=@microsoft.azure/autorest.python@~4.0.71', '--version=preview']' returned non-zero exit status 1.

Traceback (most recent call last):

Traceback (most recent call last):

  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main

    "__main__", mod_spec)

  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code

    exec(code, run_globals)

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/__main__.py", line 3, in <module>

    main(sys.argv)

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkMain.py", line 121, in main

    args.autorest_bin)

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkMain.py", line 70, in generate_sdk

    sdk_repo, temp_dir, autorest_bin)

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkNewCLI.py", line 192, in build_libraries

    autorest_bin


  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkNewCLI.py", line 148, in build_project

    autorest_bin)

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/autorest_tools.py", line 117, in generate_code

    execute_simple_command(cmd_line, cwd=str(input_path))

  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/autorest_tools.py", line 143, in execute_simple_command
```

rahuls-microsoft commented:
Is this error due to system failure-
INFO:swaggertosdk.autorest_tools:https://aka.ms/autorest
INFO:swaggertosdk.autorest_tools:   Loading AutoRest core      '/root/.autorest/@microsoft.azure_autorest-core@2.0.4390/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4390)
INFO:swaggertosdk.autorest_tools:   Installing AutoRest extension '@microsoft.azure/autorest.python' (~4.0.71)
INFO:swaggertosdk.autorest_tools:   Installing AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.44)
INFO:swaggertosdk.autorest_tools:FATAL: System.InvalidOperationException: Sequence contains no elements
INFO:swaggertosdk.autorest_tools:   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
INFO:swaggertosdk.autorest_tools:   at AutoRest.Python.Model.MethodPy.BuildInputMappings() in /home/vsts/work/1/s/src/vanilla/Model/MethodPy.cs:line 648
INFO:swaggertosdk.autorest_tools:   at AutoRest.Python.azure.Templates.AzureLongRunningMethodTemplate.<ExecuteAsync>d__1.MoveNext() in /home/vsts/work/1/s/src/obj/Razor/azure/Templates/AzureLongRunningMethodTemplate.cshtml:line 17
INFO:swaggertosdk.autorest_tools:--- End of stack trace from previous location where exception was thrown ---

Please let me know, how can I address this issue?

raych1 commented:
@zikalino, @NullMDR , can you have a look at the SDK errors?

NullMDR commented:
@zikalino Please help to check this autorest python error

jhendrixMSFT reviewed (changes requested):
I have more pending renames for operation IDs however I'm confused on your API usage.  A lot of the summary/descriptions are the same making the APIs look identical.  E.g. what's the difference between `QuotaRequest_Create` and `QuotaRequests_Create`?

rahuls-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,961 @@
+				],
+				"summary": "For the specified subscription, gets the Auto Quota Increase enrollment status.",
+				"description": "Gets the Auto Quota Increase enrollment details for the specified subscription.",
+				"operationId": "AutoQuotaIncreaseStatus_Get",
```
1. what's the difference between QuotaRequest_Create and QuotaRequests_Create? - QuotaRequest_Create is a quota increase for 1 resource Only.  QuotaRequests_Create - is a quota request for multiple resources.
2. I will change it to - AutoQuotaIncrease_GetProperties, AutoQuotaIncrease_Put (or should it be - AutoQuotaIncrease_PutProperties?).

rahuls-microsoft replied:
Will update the Put OperationId to  AutoQuotaIncrease_SetProperties.

rahuls-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,961 @@
+				],
+				"summary": "For the specified subscription, sets the Auto Quota Increase enrollment properties.",
+				"description": "Sets the Auto Quota Increase enrollment properties for the specified subscription.",
+				"operationId": "AutoQuotaIncrease_Put",
```
Will make it AutoQuotaIncrease_SetProperties.

rahuls-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,961 @@
+				},
+				"summary": "For the specified location and resource Provider, gets the current quota requests under the subscription over the time\n period of one year ago from now to one year back.",
+				"description": "For the specified location and Resource provider gets the current quota requests under the subscription over the time period of one year ago from now to one year back. oData filter can be used to select quota requests.",
+				"operationId": "QuotaRequests_listStatus",
```
Will update it to - QuotaRequests_ListStatus

rahuls-microsoft replied:
Does this request url looks fine-
/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/providers/{providerId}/locations/{location}/serviceLimits/requests.
It has 2 plural serviceLimits/requests to list all the requests.

rahuls-microsoft replied:
Also in the response for - /subscriptions/{subscriptionId}/providers/Microsoft.Capacity/providers/{providerId}/locations/{location}/serviceLimits

Should we include "requests" {"path to the requests" call - /subscriptions/{subscriptionId}/providers/Microsoft.Capacity/providers/{providerId}/locations/{location}/serviceLimits/requests?

To denote the path to get requests or the API definition and operations API good enough?
/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/providers/{providerId}/locations/{location}/serviceLimits/requests

rahuls-microsoft replied:
Or should we use -
serviceLimits/request, which Lists all the request?

Similar to  - /artifactTypes/VMExtension - https://docs.microsoft.com/en-us/rest/api/azurestack/vmextensions/list

jhendrixMSFT replied:
I'm not sure, somebody from ARM will have to answer that I think.  @KrisBash is the next on-call and hopefully can answer/finish reviewing.

rahuls-microsoft replied:
Updated it to /serviceLimts and /serviceLimitsRequests as Per suggestion from ARM team.

rahuls-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,961 @@
+				},
+				"summary": "For the specified location, gets the Quota request status by requestId.",
+				"description": "Gets the QuotaRequest details and status by the quota requestId for the resources for the resource provider at a specific location. The requestId is returned as response to the Put requests for serviceLimits.",
+				"operationId": "QuotaRequestStatus_Get",
```
will update it to - QuotaRequests_GetStatus

rahuls-microsoft committed:
1. Updating operations ID as suggested.
2. Updated summary to clarify one resource vs multiple reources.
3. Updated autoQuota Increase data model and examples.

rahuls-microsoft committed:
Updated examples.

rahuls-microsoft committed:
Fixinf spelling.

rahuls-microsoft committed:
1. Changing the enums to OnSuccess, OnFailure.

rahuls-microsoft commented:
Is there anything else outstanding, which need to be addressed?
Thanks
Rahul


Thanks.

Rahul

Sorry for fat fingers.

Sent from
Outlook<https://aka.ms/qtex0l> for iOS
________________________________
From: Joel Hendrix <notifications@github.com>
Sent: Wednesday, September 4, 2019 4:07:28 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Rahul Singh <rahul.singh@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Adding Quota APIs to Capacity RP. (#6906)


@jhendrixMSFT commented on this pull request.

________________________________

In specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6906%23discussion_r321011410&data=02%7C01%7Crahul.singh%40microsoft.com%7Cafd414d4a11740815a5508d7318ca8e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637032352508504188&sdata=57yB5rg9ZSP42gThoAiqcISEJ3%2BsQ86Yw3Iq5f3eWo0%3D&reserved=0>:

> +                             }
+                       }
+               },
+               "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/providers/{providerId}/locations/{location}/serviceLimits/requests": {
+                       "get": {
+                               "tags": [
+                                       "QuotaRequests"
+                               ],
+                               "x-ms-examples": {
+                                       "QuotaRequestHistory": {
+                                               "$ref": "./examples/getQuotaRequestsHistory.json"
+                                       }
+                               },
+                               "summary": "For the specified location and resource Provider, gets the current quota requests under the subscription over the time\n period of one year ago from now to one year back.",
+                               "description": "For the specified location and Resource provider gets the current quota requests under the subscription over the time period of one year ago from now to one year back. oData filter can be used to select quota requests.",
+                               "operationId": "QuotaRequests_listStatus",


I'm not sure, somebody from ARM will have to answer that I think. @KrisBash<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FKrisBash&data=02%7C01%7Crahul.singh%40microsoft.com%7Cafd414d4a11740815a5508d7318ca8e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637032352508514183&sdata=w0sSAQiUKMU%2FgRuPUaldCg9is%2Bcyb%2FKWTkw5DmXfc6E%3D&reserved=0> is the next on-call and hopefully can answer/finish reviewing.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6906%3Femail_source%3Dnotifications%26email_token%3DAMYYUHDGCTSCCJC27TRW5NDQIA5TBA5CNFSM4IKOH2T2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCDWJTPQ%23discussion_r321011410&data=02%7C01%7Crahul.singh%40microsoft.com%7Cafd414d4a11740815a5508d7318ca8e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637032352508524181&sdata=hJtLXTMNXgEmq6UTbou9%2FqFXMmSmMn405SqLvSXlET0%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMYYUHCWQHBQCVEYJVJBRYTQIA5TBANCNFSM4IKOH2TQ&data=02%7C01%7Crahul.singh%40microsoft.com%7Cafd414d4a11740815a5508d7318ca8e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637032352508524181&sdata=6MqvSOkYPV5mFYw%2BmDhXbzN04eWszDIWXVTydRrrjU0%3D&reserved=0>.

jhendrixMSFT commented:
Yes ARM needs to sign off on the design, once that's complete I will complete the review from the SDK perspective.

rahuls-microsoft committed:
1. Changing the serviceLimits/requests to /quotaRequests.
The requests cannot be child resource of /serviceLimits.

rahuls-microsoft committed:
1. Updating the quotaRequests to serviceRequests.
2. Updating the examples.

rahuls-microsoft committed:
Changing the serviceRequest to serviceLimitsRequests, as suggested.

rahuls-microsoft committed:
Changing emailActions to emailAction.

rahuls-microsoft committed:
1. Adding eTag header.
2. Adding oData $metadata.

rahuls-microsoft committed:
1. Making etag required parameter for put.
2. Updating metadata example.

rahuls-microsoft committed:
Updating the name to - "name": "Automatic Quota Increase",

rahuls-microsoft committed:
Adding $skipToken parameter.

rahuls-microsoft committed:
Adding If-Match to the put and patch examples for quota.

rahuls-microsoft committed:
1. Updating the Examles based on the model.

rahuls-microsoft committed:
Updating examples based on model.

rahuls-microsoft committed:
Fixing examples' model for autoQuotaIncrease.

rahuls-microsoft committed:
Fixing examples Model errors.

rahuls-microsoft committed:
Fixing recipients.

rahuls-microsoft committed:
Changing the AutoQuotaIncrease Put data model.

rahuls-microsoft committed:
Fixing json.

rahuls-microsoft committed:
Fixing example.

rahuls-microsoft committed:
Removing $metadata api.

rahuls-microsoft committed:
1. Adding subRequestId for subrequests.
2. Updating examples with subRequestId.

ravbhatnagar commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1045 @@
+						"in": "body",
+						"description": "Quota requests payload.",
+						"required": true,
+						"schema": {
```
request body and response body must use the same model. Any properties which is not settable by the end customer can be marked as readonly

rahuls-microsoft replied:
The Response body has the proviionState field inside the properties. We need to communicate the status of the quota request. The fields have bee marked as read-only.

rahuls-microsoft replied:
Based on - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources 
Changed to 201 with response in following format as documented  -
{
  "location": "North US",
  "tags": {
    "key1": "value 1",
    "key2": "value 2"
  },

  "properties": {
    "provisioningState": "Succeeded",
    "comment": "Resource defined structure"
  },
}

sanjaiganesh replied:
@rahuls-microsoft  it is not addressed, as request and response is not matching. Request body, 200 & 201 response must have the same payload. why do we have three definitions ?

What is the async pattern used for 201 ? If the service is using provisioningState based polling, then please update description .(also same as request body).

Mark provisioningState as readonly property

rahuls-microsoft replied:
I will update the definitions to match and add  provisioningState as readonly property

rahuls-microsoft replied:
Updated it have same type for request and response.

ravbhatnagar commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1045 @@
+					},
+					"202": {
+						"description": "The request is accepted and is being processed. Please use requestId to check the status.",
+						"schema": {
```
this should not have any response body. The location header to track the status of this operation is provided as a header.

rahuls-microsoft replied:
Changed it to 201 response as @KrisBash suggested.

ravbhatnagar commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1045 @@
+					"nextLinkName": "nextLink"
+				}
+			},
+			"put": {
```
what is this PUT? the API signature is not correct. You have a PUT on serviceLImits/{name} what is the intent of this PUT?

rahuls-microsoft replied:
This to put quota request for one resource.
Example- Put - …/Microsoft.Cpmpute/eastus/standardNDSFamily

 {
				"limit": 20,
				"unit": "Count",
				"currentValue":10,
				
This will be used by BluePrint, as you suggested earlier.

rahuls-microsoft replied:
Does it sound fine?

ravbhatnagar commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1045 @@
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/autoQuotaIncrease": {
```
what is autoQuotaIncreate?

rahuls-microsoft replied:
AutoQuotaIncrease is a feature for a subscription, where user can opt in to automatically increase the quota, as needed.

ravbhatnagar reviewed (commented):
@rahuls-microsoft - the APIs still look bit fuzzy. please schedule a meeting with armapireview to close on this. this is a very old PR, not sure if this is the latest. Some concepts like autoincreatequota seem new and weren't discussed during our prior meetings.

yungezz commented:
hi @rahuls-microsoft are you still on this PR?

rahuls-microsoft commented:
Hi,
We would like to complete the API review soon.

I worked with @Kristopher Bash<mailto:krbash@microsoft.com> to update the APIs for /ServiceLimits and /ServiceLimits and /serviceLimitsRequests.
So, that should be good.

We added AutoQuotaIncrease Request API to enable the AutoQuotaIncrease functionality, which is being added to Azure portal for a subscription.
This will be mainly used by Azure portal, as it is one time setting to enable autoQuotaIncrease or when a new subscription is created using Arm/bluePrint.
I will schedule meeting with @Gaurav Bhatnagar<mailto:Gaurav.Bhatnagar@microsoft.com> to go over the details and questions.


Thanks.
Rahul.

From: Yunge Zhu <notifications@github.com>
Sent: Thursday, September 26, 2019 12:51 AM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Rahul Singh <rahul.singh@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Adding Quota APIs to Capacity RP. (#6906)


hi @rahuls-microsoft<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frahuls-microsoft&data=02%7C01%7Crahul.singh%40microsoft.com%7C9beafa4883c4401111e008d7425640b1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637050810537436794&sdata=KROK1K5%2BTuU3u4V1Vqsi09YQhUirntMLTzvvrGIAyhw%3D&reserved=0> are you still on this PR?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6906%3Femail_source%3Dnotifications%26email_token%3DAMYYUHA7AOS7STQZLSNFFY3QLRSVXA5CNFSM4IKOH2T2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7UVDQQ%23issuecomment-535384514&data=02%7C01%7Crahul.singh%40microsoft.com%7C9beafa4883c4401111e008d7425640b1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637050810537446791&sdata=AYr5VOxIyUiAzEMLMvhhg9gm%2FlxOgLCndB11gpC6Zj0%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMYYUHCD45CZ43P6XPHC2IDQLRSVXANCNFSM4IKOH2TQ&data=02%7C01%7Crahul.singh%40microsoft.com%7C9beafa4883c4401111e008d7425640b1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637050810537446791&sdata=TSk6bOZWf8B9JlJ29XuZPt9nC3viEYvmvrtt1HVWXUg%3D&reserved=0>.

yungezz commented:
thanks for the update @rahuls-microsoft . sound good.

rahuls-microsoft committed:
1. Updating long running task response based on comments.
2. Updating Auto Quota Increase request.
Auto Quota increase is a functionality/service, where a customer can opt in for - service monitoring the usages and quota value and automatically increase the quota, when a threshold is reached.

rahuls-microsoft committed:
Fixing builderror camel case.
Fixing example.

rahuls-microsoft committed:
Changing "operationId": "AutoQuotaIncrease_Create"

rahuls-microsoft committed:
Changing the field names to lower case.

rahuls-microsoft committed:
Changing case.

rahuls-microsoft committed:
Changing Case.

rahuls-microsoft committed:
Updating Auto Quota Increase schema.

rahuls-microsoft committed:
Fixing AQI examples.

rahuls-microsoft committed:
Removing readonly properties from Set requests.

rahuls-microsoft committed:
Removing unused model.

rahuls-microsoft committed:
changing to lowercase

rahuls-microsoft committed:
Fixing schema.

rahuls-microsoft committed:
Fixing nested refs.

rahuls-microsoft committed:
Changing type to object.

rahuls-microsoft committed:
example fix.

rahuls-microsoft committed:
Updated the Put schema and examle, as per discussion.

rahuls-microsoft committed:
Updated Examples and formatting.

rahuls-microsoft committed:
Fixing errors/warnings.

rahuls-microsoft committed:
Put Response fix.

rahuls-microsoft committed:
Model Fix.

rahuls-microsoft committed:
Model Fixes.

rahuls-microsoft committed:
removing "x-ms-azure-resource": true from request.

rahuls-microsoft committed:
Updating response schema for Put.

rahuls-microsoft committed:
Put examples fixing.

rahuls-microsoft committed:
Adding type= object for top level models.

rahuls-microsoft committed:
Example fix.

rahuls-microsoft committed:
Quota request response updates.

rahuls-microsoft committed:
Example Fix.

rahuls-microsoft committed:
Model fix.

rahuls-microsoft committed:
Changing model.

rahuls-microsoft committed:
Example Fix.

rahuls-microsoft committed:
One SKU Put quota request update.

rahuls-microsoft committed:
Fixing Put requests.

rahuls-microsoft committed:
Example updates.

rahuls-microsoft committed:
Making currentValue readonly.

rahuls-microsoft committed:
Remove currentValue from quota update response.

rahuls-microsoft committed:
Model Fix for put.

rahuls-microsoft committed:
Fixing Patch example.

rahuls-microsoft commented:
All the open items addressed. Please review it.

rahuls-microsoft committed:
Making proerties read-only.

rahuls-microsoft committed:
Removing readonly properties from resource name and unit

rahuls-microsoft committed:
removing ready-only from unit.

rahuls-microsoft commented:
@KrisBash Please review the PR and let me know, if any other changes needed. Thanks.

rahuls-microsoft committed:
1. Removing collection Put and Patch.
2. Adding patch for single resource.

KrisBash reviewed (approved):
Signing off. Thanks for the updates

rahuls-microsoft committed:
1. Making Put request and response body same.
2. Making provisioningState as readonly property.
3. Updated Examples accordingly.

rahuls-microsoft committed:
Updating Patch operation name to _Update.

rahuls-microsoft committed:
Model Update.

rahuls-microsoft committed:
Fixing examples.

rahuls-microsoft committed:
Fixing examples.

rahuls-microsoft committed:
1. Fixing Examples.

rahuls-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1270 @@
+					"nextLinkName": "nextLink"
+				}
+			},
+			"put": {
```
As we discussed there is Get for /serviceLimits/{resourceName}, /serviceLimits.

rahuls-microsoft replied:
Line 37 quota.json.

rahuls-microsoft commented:
All comments resolved. Please take a look and give a final approval. Thanks.

jhendrixMSFT commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1270 @@
+				],
+				"summary": "Gets the current quota limit and usages for the resource provider for the specified location for the specific resource in the parameter.",
+				"description": "This API gets the current quota limit and usages for the specific resource for resource provider for the specified location. This response can be used to submit quotaRequests.",
+				"operationId": "Quota_listStatus",
```
`Quotas_GetStatus`

rahuls-microsoft replied:
Updated as suggested.

jhendrixMSFT commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1270 @@
+				],
+				"summary": "Submits a Quota Request for a resource provider at the specified location for the specific resource in the parameter.",
+				"description": "Submits Quota change request for a resource provider for the specified location for the specific resource in the parameter. To use, first make a Get request to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which requires an update to their quotas, update their limit fields i ...
+				"operationId": "QuotaRequest_Create",
```
`Quotas_Create` so that all operations under this path are grouped together.

rahuls-microsoft replied:
Updated as suggested.

jhendrixMSFT commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1270 @@
+				],
+				"summary": "Submits a Quota Request for a resource provider at the specified location for the specific resource in the parameter.",
+				"description": "Submits Quota change request for a resource provider for the specified location for the specific resource in the parameter. To use, first make a Get request to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which requires an update to their quotas, update their limit fields i ...
+				"operationId": "QuotaRequest_Update",
```
`Quotas_Update` so that all operations under this path are grouped together.

rahuls-microsoft replied:
Updated as suggested.

jhendrixMSFT reviewed (changes requested):
Just a few operation ID to fix so all the logical ones are grouped together.

rahuls-microsoft committed:
Updating operation id as suggested.

jhendrixMSFT commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -0,0 +1,1270 @@
+				],
+				"summary": "Gets the current quota limit and usages for the resource provider for the specified location for the specific resource in the parameter.",
+				"description": "This API gets the current quota limit and usages for the specific resource for resource provider for the specified location. This response can be used to submit quotaRequests.",
+				"operationId": "Quota_GetStatus",
```
Missing 's', i.e. `Quotas_GetStatus`

rahuls-microsoft replied:
Updated to Quotas_GetStatus

rahuls-microsoft committed:
Updated Operation Id: "operationId": "Quotas_GetStatus",

rahuls-microsoft committed:
Fixing issue - The PUT/GET/PATCH operations must have same schema response.

rahuls-microsoft committed:
1. Adding id, name, type properties in reponse example.
2. Making ResourceType property setable for Put,Patch request.

jhendrixMSFT commented:
@rahuls-microsoft thanks for working through all the issues.  Is this functionality live?

rahuls-microsoft commented:
Not yet. We planning preview in December.
Thanks
Rahul


Thanks.

Rahul

Sorry for fat fingers.

Sent from
Outlook<https://aka.ms/qtex0l> for iOS
________________________________
From: Joel Hendrix <notifications@github.com>
Sent: Thursday, October 17, 2019 12:32:46 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Rahul Singh <rahul.singh@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Adding Quota APIs to Capacity RP. (#6906)


@rahuls-microsoft<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frahuls-microsoft&data=02%7C01%7Crahul.singh%40microsoft.com%7C1658e5c85a8648d16bb808d75338ca4e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069375683995894&sdata=N48Ql7TZVIx%2BxFBnhNYFApWSnOfOy5jVv0jIEOr0eUk%3D&reserved=0> thanks for working through all the issues. Is this functionality live?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6906%3Femail_source%3Dnotifications%26email_token%3DAMYYUHHXBZEMSKI6YGGFKG3QPC4V5A5CNFSM4IKOH2T2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBRIJKY%23issuecomment-543327403&data=02%7C01%7Crahul.singh%40microsoft.com%7C1658e5c85a8648d16bb808d75338ca4e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069375683995894&sdata=Pg8%2BiQpDXmSTxvp1qqPGlXyzzt7ug9ObK0klL9O3XDM%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMYYUHHGXDPDXKIXQLD5AN3QPC4V5ANCNFSM4IKOH2TQ&data=02%7C01%7Crahul.singh%40microsoft.com%7C1658e5c85a8648d16bb808d75338ca4e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069375684005851&sdata=Z4xVTFw4lMaUL4nC0Se6Q54u9lEJN2gPtEBpH1Rs0Dw%3D&reserved=0>.

rahuls-microsoft committed:
Updating the samples

rahuls-microsoft committed:
Updating the samples

rahuls-microsoft committed:
Revert commit.

rahuls-microsoft committed:
Merge branch 'rahuls-quota-spec' of https://github.com/rahuls-microsoft/azure-rest-api-specs into rahuls-quota-spec

* 'rahuls-quota-spec' of https://github.com/rahuls-microsoft/azure-rest-api-specs:
  Updating the samples

rahuls-microsoft committed:
Formatting the json.
Example spelling correction.

rahuls-microsoft committed:
1. Chaging resource name from Providers to resourceProviders, as suggested by ARM team.
2. Updating examples.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Fixing jSon.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Adding patch API.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Adding resource name to the example.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Updating patch example.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Updating Patch example.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Patch example fix.

rahuls-microsoft committed:
Remove unit from request example.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Removing readonly property Unit from request.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Updating request format.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Changing the title to match the reservation - Azure Reservation API

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Changing the tile back to - Microsoft.Capacity

AutorestCI commented:
/azp run automation - sdk

jhendrixMSFT commented:
When you say "private preview" does that mean only whitelisted subscriptions will have access to this API version?

rahuls-microsoft commented:
That’s right. We are working with couple of internal Microsoft teams to onboard.

Thanks.
Rahul.

From: Joel Hendrix <notifications@github.com>
Sent: Wednesday, January 8, 2020 3:05 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Rahul Singh <rahul.singh@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Adding Quota APIs to Capacity RP. (#6906)


When you say "private preview" does that mean only whitelisted subscriptions will have access to this API version?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6906%3Femail_source%3Dnotifications%26email_token%3DAMYYUHBZL5WSG4XNWHYAAXLQ4ZL2JA5CNFSM4IKOH2T2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIOJ3IQ%23issuecomment-572300706&data=02%7C01%7Crahul.singh%40microsoft.com%7Cec7c1eb000274591f3d608d7948f35a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637141215114916905&sdata=Vy9XFpP3nMaaxshTw%2BVFwpxFRlz5aFmaFbgjSwwwJ9c%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMYYUHCZWFCAV62IVRIYUCDQ4ZL2JANCNFSM4IKOH2TQ&data=02%7C01%7Crahul.singh%40microsoft.com%7Cec7c1eb000274591f3d608d7948f35a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637141215114926860&sdata=SyEuo%2F1JciF%2B1OjX5qzEzWzK2ZXc%2FpQrvdzsTwOsr%2Fo%3D&reserved=0>.

raych1 commented:
/azp run

rahuls-microsoft committed:
Removing Readonly roperty from the quota request.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Including stable version of Reservation API for tag-
 tag: package-preview-2019-07-19.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Matching the title with the released version of reservation Api.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Including the reservation latest swagger file.
Matching the Title to Microsoft.Capacity.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Reverting the title back to Azure Reservation API

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Running json formatter.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Formatting json from vscode.

The prettier didn't work! Used vscode to format the json.

Error:
D:\src\azureAPI\azure-rest-api-specs>npm run prettier -- --write "specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/**/*.json"
npm ERR! missing script: prettier

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Running json Prettier.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Running formatting from visul studio.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Running prettier check.
Output shows no Issues-
D:\src\azureAPI\azure-rest-api-specs>prettier -c specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/*.json
Checking formatting...
All matched files use Prettier code style!

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
Hi, I am constantly getting PrettierCheck failure from Azure API build.

When I run it locally it passes-
D:\src\azureAPI\azure-rest-api-specs>prettier -c specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/*.json
Checking formatting...
All matched files use Prettier code style!

Fails on Azure build-
https://dev.azure.com/azure-sdk/public/_build/results?buildId=240008&view=logs&j=b56e9678-4dad-5642-289c-5530cc4dad11&t=9e18c5d1-dc43-58a1-0344-6ef37d05bd7f
Checking formatting...
specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json
Code style issues found in the above file(s). Forgot to run Prettier?

Any idea - How to fix it?
I ran Prettier.

Thanks.
Rahul.

NullMDR commented:
@rahuls-microsoft Please merge from latest master and try again

rahuls-microsoft commented:
> @rahuls-microsoft Please merge from latest master and try again

https://github.com/Azure/azure-rest-api-specs/compare/master...rahuls-microsoft:master?diff=split

How do I merge from - 
![image](https://user-images.githubusercontent.com/53578268/72871313-56af0a00-3c9f-11ea-8da8-a1c3b2dd6f55.png)

Azure/azure-rest-api-specs to rahuls-microsoft/azure-rest-api-specs 

Thanks.

rahuls-microsoft committed:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

rahuls-microsoft committed:
Merge branch 'master' into rahuls-quota-spec

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Running Prettier.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
Can we merge the API? We are stating preview at 1/31/20.
Thanks.

jhendrixMSFT commented:
The CI needs to pass before this can be merged.

jhendrixMSFT commented:
/azp run automation

rahuls-microsoft commented:
> No pipelines are associated with this pull request.

What's required from my side?
How do I trigger - /azp run automation

jhendrixMSFT commented:
I don't know why it says "no pipeline are associated", hopefully @NullMDR can help with that.

NullMDR commented:
/azp run automation - sdk

NullMDR commented:
@jhendrixMSFT That's kind of known bug I'll fix later. Currently you can use "/azp run automation - sdk" to workaround this issue

zikalino commented:
/azp run automation - sdk
